pkg/common/cns-lib/volume/manager.go

```diff
@@ -315,23 +315,59 @@ func (m *defaultManager) createVolumeWithImprovedIdempotency(ctx context.Context
 		}
 	}
 
-	// Get the taskInfo.
-	taskInfo, err := cns.GetTaskInfo(ctx, task)
-	if err != nil || taskInfo == nil {
-		log.Errorf("failed to get taskInfo for CreateVolume task with err: %v", err)
-		if err != nil {
-			faultType = ExtractFaultTypeFromErr(ctx, err)
-		} else {
-			faultType = csifault.CSITaskInfoEmptyFault
+	taskInfo, err := task.WaitForResult(ctx, nil)
+	if err != nil {
+		// WaitForResult can fail for many reasons, including:
+		// - Task was completed and deleted from VCDB.
+		// - CNS restarted and marked "InProgress" tasks as "Failed".
```

Comment:
It could fail with create workflow errors in CNS and FCD layers right? If yes, can you add that condition too to this comment so that we know we are handing all errors in the same way?

Reply:
done